### PR TITLE
Sunset Site Assembler: Remove all CTAs

### PR DESCRIPTION
**Pull request by taipeicoder**
Related to https://github.com/Automattic/dotcom-forge/issues/10066

## Proposed Changes

This PR removes the Site Assembler CTAs that are now deprecated. These CTAs affect three areas:
- Theme Showcase
- Onboarding Design Picker
- Onboarding Design Choice

## Why are these changes being made?

Sunset Site Assembler.

## Testing Instructions

Smoke-test the three areas mentioned above and ensure that they still work as before, with no regression issues.
Since the Site Assembler CTAs are already disabled in production, this PR shouldn't introduce any UI changes.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [x] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/themes/theme-showcase.jsx [hunk 1]
```diff
@@ -388,40 +384,6 @@ class ThemeShowcase extends Component {
...
-		const { isLoggedIn } = this.props;
-
-		recordTracksEvent( 'calypso_themeshowcase_pattern_assembler_top_button_click', {
-			is_logged_in: isLoggedIn,
-		} );
-
-		this.onDesignYourOwnCallback();
-	};
-
-	onDesignYourOwnCallback = () => {
```
Comment (fushar) [inline client/my-sites/themes/theme-showcase.jsx:401]: There are still multiple occurrences of `this.onDesignYourOwnCallback` in this file; let's also clean them up.

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-choices/index.tsx [hunk 2]
```diff
@@ -36,9 +31,6 @@ const DesignChoicesStep: Step = ( { navigation, flow, stepName } ) => {
 	);
 
 	const { isEligible, isLoading } = useIsBigSkyEligible();
-
-	const isSiteAssemblerEnabled = useIsSiteAssemblerEnabled();
```
Comment (fushar) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-choices/index.tsx:40]: We can now clean up `useIsSiteAssemblerEnabled()` definition (`client/data/site-assembler/index.ts`)

Comment (taipeicoder) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-choices/index.tsx:40]: Good catch 🙂

**Review by fushar — APPROVED** [review]
Did not find any regression when testing DP and theme showcase. Themes can still be activated successfully 👍

**Pull request merged**